src/cmakeTaskProvider.ts

```diff
@@ -373,10 +373,17 @@ export class CustomBuildTaskTerminal implements vscode.Pseudoterminal, proc.Outp
             const result: proc.ExecutionResult = await proc.execute(cmakePath, args, this, this.options).result;
             if (result.retc) {
                 this.writeEmitter.fire(localize("build.finished.with.error", "{0} finished with error(s).", taskName) + endOfLine);
-            } else if (result.stderr && !result.stdout) {
-                this.writeEmitter.fire(localize("build.finished.with.warnings", "{0} finished with warning(s).", taskName) + endOfLine);
-            } else if (result.stdout && result.stdout.includes("warning")) {
+                if (doCloseEmitter) {
```

Comment:
This shouldn't be done here.  There is a `this.closeEmitter.fire` call a few lines down where you should pass in `result.retc ?? 0` instead.